spec/index.md

```diff
@@ -651,6 +680,50 @@ The functionality exposed via the [The IC management canister](#ic-management-ca
 
 :::
 
+### Request: Sync Call {#http-sync-call}
+A synchronous update call, or "call and await", is a type of update [call](#http-call) where the replica will wait with replying to the user until the call has been processed and the result has been added to the replicated state. This means the user will receive a certified result of the call, and thus __do not need to poll__ [`read_state`](#http-read-state) to determine the status of the call.
+In order to make a synchronous update call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/sync_call`. The request body consists of an authentication envelope with a `content` map with the following fields:
+
+-   `request_type` (`text`): Always `sync-call`
+
+-   `sender`, `nonce`, `ingress_expiry`: See [Authentication](#authentication)
+
+-   `canister_id` (`blob`): The principal of the canister to call.
+
+-   `method_name` (`text`): Name of the canister method to call
+
+-   `arg` (`blob`): Argument to pass to the canister method
+
+The HTTP response to this request can have the following responses:
+
+-   202 HTTP status with a non-empty body. Implying the request was processed.
```

Comment:
202 suggests that the request has been accepted for further processing which is not the case here - the status code 200 should be returned

Reply:
Good point. I changed it to `201` for the cases where a certification is returned. 

We can keep `202` for cases where the user needs to fall back to polling, for example if we time out the requests because no response has been generated within some time threshold.

Reply:
Why not 200?

Reply:
> We can keep 202 for cases where the user needs to fall back to polling, for example if we time out the requests because no response has been generated within some time threshold.

That makes sense!

Reply:
Thanks for the feedback!

>I do not think HTTP 201 is appropriate here. While the request method is POST, it is not true that the request "has led to the creation of a resource".

Can't we argue that response status + Certification that's returned (which is added to the replicated state) is a "creation of a resource"?

I agree that making `200` generic over `reply` and `reject` would make things more consistent with how canister calls work internally. However, reserving `200` for `reject` makes the API of our sync endpoint more consistent with the async endpoint for the user.

I don't have a strong opinion about either option. We can go with a generic `200` if you think it makes more sense.

CC: @rumenov